plutus-pab-executables/pab-cli/App.hs

```diff
@@ -0,0 +1,468 @@
+            liftIO $ runAsyncActions [ startChainIndexAction
+                                     , startCardanoNodeAction
+                                     , startCardanoWalletAction
+                                     ]
+        Just pabOpts -> do
+            -- Generate base PAB configuration and modify the node and wallet
+            -- server config
+            let pabServerConfig = (pabWebserverBaseConfig nodeServerConfig pabOpts appOptsChainIndexOpts)
+                    { walletServerConfig = walletConfig
+                    , nodeServerConfig = nodeServerConfig
+                    }
+
+            let startPabWebserverAction = startPabWebserver pabOpts pabServerConfig
+
+            liftIO $ runAsyncActions [ startChainIndexAction
+                                     , startCardanoNodeAction
+                                     , startPabWebserverAction
+                                     , startCardanoWalletAction
+                                     ]
+
+runAppCommand (NodeRemoteWalletCommand network pabOptsM) = do
+    appOpts@AppOpts { appOptsNodeOpts = NodeOpts { nodeOptsOutputDir }
+            , appOptsChainIndexOpts
+            } <- ask @AppOpts
+    -- Fetch node config files depending on network and save to node output directory.
+    liftIO $ fetchNodeConfigFiles network nodeOptsOutputDir
+
+    nodeServerConfig <- getCardanoNodeConfig network nodeOptsOutputDir
+    let networkId = unNetworkIdWrapper $ pscNetworkId nodeServerConfig
+        networkName = getNetworkName networkId
+        startChainIndexAction =
+            startChainIndex networkId appOptsChainIndexOpts (appOptsNodeOpts appOpts)
+        startCardanoNodeAction = startCardanoNode networkName (appOptsNodeOpts appOpts)
+
+    case pabOptsM of
+        Nothing ->
+            liftIO $ runAsyncActions [ startChainIndexAction
+                                     , startCardanoNodeAction
+                                     ]
+        Just pabOpts -> do
+            let pabServerConfig = (pabWebserverBaseConfig nodeServerConfig pabOpts appOptsChainIndexOpts)
+                    { walletServerConfig = RemoteWalletConfig
+                    , nodeServerConfig = nodeServerConfig
+                    }
+
+            let startPabWebserverAction = startPabWebserver pabOpts pabServerConfig
+
+            liftIO $ runAsyncActions [ startChainIndexAction
+                                     , startCardanoNodeAction
+                                     , startPabWebserverAction
+                                     ]
+
+-- | Run list of actions asynchroniously and wait until any one of them finishes.
+runAsyncActions :: [IO ()] -> IO ()
+-- runAsyncActions ios = do
+--     asyncs <- mapM (\io -> withAsync io pure) ios
+--     print $ length asyncs
+--     void $ waitAny asyncs
+runAsyncActions ios = void $ go ios []
+    where go [] asyncs = waitAny asyncs
+          go (x:xs) asyncs = do
+              withAsync x $ \a -> go xs (a : asyncs)
```

Comment:
The `Alternative` instance of `Concurrently` does `race` which I think is what you want here, so:
```haskell
runAsyncActions = ala (Alt . Concurrently) foldMap
```

Reply:
Nice!